### PR TITLE
More-objects migration + Adding equals,hashCode,toString to various classes

### PR DESCRIPTION
**Pull request by Jw0x47**
Implements the changes from #361 on top of the recent set of changes by gchomatas. 

FYI @stevegutz

### CODE REVIEW

**Comment by Jw0x47** [comment]
This has been merged to staging.

**Comment by Jw0x47** [comment]
06c7ae0 Would have ended up conflicting heavily with this change, so I added it here for simplicity's sake.

File: BlazarBase/src/main/java/com/hubspot/blazar/external/models/singularity/BuildCGroupResources.java [hunk 1]
```diff
@@ -39,4 +42,30 @@ public BuildCGroupResources add(BuildCGroupResources other) {
...
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    BuildCGroupResources that = (BuildCGroupResources) o;
+    return Double.compare(that.cpus, cpus) == 0 &&
```
Comment (stevie400) [inline BlazarBase/src/main/java/com/hubspot/blazar/external/models/singularity/BuildCGroupResources.java]: You can just use `Objects.equals` for all fields and call it a day.

**Comment by Jw0x47** [comment]
This has been merged to staging, staging environment looks 👍

**Comment by gchomatas** [comment]
🚢

**Pull request merged**